tests/sos_tests.py

```diff
@@ -0,0 +1,464 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+
+from avocado.core.exceptions import TestSkipError
+from avocado import Test
+from avocado.utils import archive, process
+from fnmatch import fnmatch
+
+import glob
+import json
+import os
+import pickle
+import re
+
+SOS_TEST_DIR = os.path.dirname(os.path.realpath(__file__))
+SOS_BIN = os.path.realpath(os.path.join(SOS_TEST_DIR, '../bin/sos'))
+
+
+def skipIf(cond, message=None):
```

Comment:
Any reason for not using [`avocado.skipIf()`](https://avocado-framework.readthedocs.io/en/86.0/api/test/avocado.html#avocado.skipIf) ?

Reply:
This test is defined in the `StageOneReportTest` class that our other tests subclass:

```python
    @skipIf(lambda x: '--no-report' in x.sos_cmd, '--no-report used in command')
    def test_html_reports_created(self):
```

The problem I was running into was that `avocado.skipIf()` was not properly inspecting the child class' `sos_cmd` attribute,  so it was not properly skipping if the `--no-report` option was being used.

If there is another way around this using the avocado provided decorator, I'm all ears.

Reply:
So no, there was no way to do that with the Avocado decorators.  But, I think it's a valid feature, so I proposed this:

https://github.com/avocado-framework/avocado/pull/4467

If you could take a quick look (or even a formal review), it'd be very much appreciated.